airbyte-integrations/connectors/source-monday/source_monday/graphql_requester.py

```diff
@@ -79,18 +79,37 @@ def _build_query(self, object_name: str, field_schema: dict, **object_arguments)
         arguments = f"({arguments})" if arguments else ""
         fields = ",".join(fields)
 
-        return f"{object_name}{arguments}{{{fields}}}"
+        if object_name in ["items_page", "next_items_page"]:
+            query = f"{object_name}{arguments}{{cursor,items{{{fields}}}}}"
+        else:
+            query = f"{object_name}{arguments}{{{fields}}}"
+        return query
 
     def _build_items_query(self, object_name: str, field_schema: dict, sub_page: Optional[int], **object_arguments) -> str:
         """
         Special optimization needed for items stream. Starting October 3rd, 2022 items can only be reached through boards.
         See https://developer.monday.com/api-reference/docs/items-queries#items-queries
+
+        Comparison of different APIs queries:
+        2023-07:
+            boards(limit: 1)         {      items(limit: 20)                 {              field1, field2, ...  }}
+            boards(limit: 1, page:2) {      items(limit: 20, page:2)         {              field1, field2, ...  }} boards and items paginations
+        2024_01:
+            boards(limit: 1)         { items_page(limit: 20)                 {cursor, items{field1, field2, ...} }}
+            boards(limit: 1, page:2) { items_page(limit: 20)                 {cursor, items{field1, field2, ...} }} - boards pagination
+                                  next_items_page(limit: 20, cursor: "blaa") {cursor, items{field1, field2, ...} }  - items pagination
+
         """
         nested_limit = self.nested_limit.eval(self.config)
 
-        query = self._build_query("items", field_schema, limit=nested_limit, page=sub_page)
-        arguments = self._get_object_arguments(**object_arguments)
-        return f"boards({arguments}){{{query}}}"
+        if sub_page:
+            query = self._build_query("next_items_page", field_schema, limit=nested_limit, cursor=f'"{sub_page}"')
+        else:
+            query = self._build_query("items_page", field_schema, limit=nested_limit, page=sub_page)
```

Comment:
Is there a reason to pass `page=sub_page` here? As `sub_page==None`, I don't see the value added and it generates a lot of questions for the reader

Reply:
it is removed